python/semantic_kernel/connectors/memory/weaviate/weaviate_collection.py

```diff
@@ -148,47 +169,137 @@ async def _inner_upsert(
+            await collection.data.delete_many(where=Filter.any_of([Filter.by_id().equal(key) for key in keys]))
+        except Exception as ex:
+            raise MemoryConnectorException(f"Failed to delete records: {ex}")
+
+    @override
+    async def _inner_search(
+        self,
+        options: VectorSearchOptions,
+        search_text: str | None = None,
+        vectorizable_text: str | None = None,
+        vector: list[float | int] | None = None,
+        **kwargs: Any,
+    ) -> KernelSearchResults[VectorSearchResult[TModel]]:
+        vector_field = self.data_model_definition.try_get_vector_field(options.vector_field_name)
+        collection: CollectionAsync = self.async_client.collections.get(self.collection_name)
+        args = {
+            "filters": _create_filter_from_vector_search_filters(options.filter),
+            "include_vector": options.include_vectors,
+            "limit": options.top,
+            "offset": options.skip,
+        }
+        if search_text:
+            results = await self._inner_vector_text_search(collection, search_text, args)
+        elif vectorizable_text:
+            results = await self._inner_vectorizable_text_search(collection, vectorizable_text, vector_field, args)
+        elif vector:
+            results = await self._inner_vectorized_search(collection, vector, vector_field, args)
+        else:
+            raise MemoryConnectorException("No search criteria provided.")
+
+        return KernelSearchResults(
+            results=self._get_vector_search_results_from_results(results.objects), total_count=len(results.objects)
+        )
+
+    async def _inner_vector_text_search(
+        self, collection: CollectionAsync, search_text: str, args: dict[str, Any]
+    ) -> Any:
+        try:
+            return await collection.query.bm25(
+                query=search_text,
+                return_metadata=MetadataQuery(score=True),
+                **args,
+            )
+        except Exception as ex:
+            raise MemoryConnectorException(f"Failed searching using a text: {ex}") from ex
+
+    async def _inner_vectorizable_text_search(
+        self,
+        collection: CollectionAsync,
+        vectorizable_text: str,
+        vector_field: VectorStoreRecordVectorField | None,
+        args: dict[str, Any],
+    ) -> Any:
+        if self.named_vectors and not vector_field:
+            raise MemoryConnectorException(
+                "Vectorizable text search requires a vector field to be specified in the options."
+            )
+        try:
+            return await collection.query.near_text(
+                query=vectorizable_text,
+                target_vector=vector_field.name if self.named_vectors and vector_field else None,
+                return_metadata=MetadataQuery(distance=True),
+                **args,
+            )
+        except Exception as ex:
+            logger.error(
+                f"Failed searching using a vectorizable text: {ex}. "
```

Comment:
nit: Put all the setup information in the README.md file under weaviate/ too so that we can have some sort reference when creating learn site docs.